src/System.IO/tests/BinaryReader/BinaryReaderTests.cs

```diff
@@ -91,5 +91,43 @@ private void ValidateDisposedExceptions(BinaryReader binaryReader)
             Assert.Throws<ObjectDisposedException>(() => binaryReader.ReadUInt32());
             Assert.Throws<ObjectDisposedException>(() => binaryReader.ReadUInt64());
         }
+
+        public class NegEncoding : UTF8Encoding
+        {
+            public override Decoder GetDecoder()
+            {
+                return new NegDecoder();
+            }
+
+            public class NegDecoder : Decoder
+            {
+                public override int GetCharCount(byte[] bytes, int index, int count)
+                {
+                    return 1;
+                }
+
+                public override int GetChars(byte[] bytes, int byteIndex, int byteCount, char[] chars, int charIndex)
+                {
+                    return -10000000;
+                }
+            }
+        }
+
+        [Fact]
```

Comment:
The BinaryReaderTests are all netstandard17 specific. It should pass all netstandard17 platforms.